realm/realm-library/src/androidTest/kotlin/io/realm/ParameterizedSetTests.kt

```diff
@@ -0,0 +1,181 @@
+ * and `unmanaged` modes.
+ *
+ * In order to streamline the testing for managed dictionaries we use Kotlin's reflection API
+ * `KFunction1` and `KFunction2`. These two methods provide access to the Java getters and setters
+ * used to work with each dictionary field.
+ */
+@RunWith(Parameterized::class)
+class ParameterizedSetTests(
+        private val tester: SetTester
+) {
+
+    /**
+     * Initializer for parameterized tests
+     */
+    companion object {
+
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0}")
+        fun testType(): List<SetTester> {
+            return SetMode.values().mapNotNull { type ->
+                when (type) {
+                    SetMode.UNMANAGED -> unmanagedSetFactory()
+                    SetMode.MANAGED -> null
+                }
+            }.flatten()
+        }
+    }
+
+    @Rule
+    @JvmField
+    val configFactory = TestRealmConfigurationFactory()
+
+    private val looperThread = BlockingLooperThread()
+
+    @Before
+    fun setUp() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().context)
+        tester.setUp(configFactory.createConfiguration(), looperThread)
+    }
+
+    @After
+    fun tearDown() {
+        tester.tearDown()
+    }
+
+    @Test
+    fun isManaged() {
+        tester.isManaged()
+    }
+
+    @Test
+    fun isValid() {
+        tester.isValid()
+    }
+
+    @Test
+    fun isFrozen() {
+        tester.isFrozen()
+    }
+
+    @Test
+    fun size() {
+        tester.size()
+    }
+
+    @Test
+    fun isEmpty() {
+        tester.isEmpty()
+    }
+
+    @Test
+    fun contains() {
+        tester.contains()
+    }
+
+    @Test
+    fun iterator() {
+        tester.iterator()
+    }
+
+    @Test
+    fun toArray() {
+        tester.toArray()
+    }
+
+    @Test
+    fun add() {
+        tester.add()
+    }
+
+    @Test
+    fun remove() {
+        tester.remove()
+    }
+
+    @Test
+    fun containsAll() {
+        tester.containsAll()
+    }
+
+    @Test
+    fun addAll() {
+        tester.addAll()
+    }
+
+    @Test
+    fun retainAll() {
+        tester.retainAll()
+    }
+
+    @Test
+    fun removeAll() {
+        tester.removeAll()
+    }
+
+    @Test
+    fun clear() {
+        tester.clear()
+    }
+
+    @Test
+    fun freeze() {
+        tester.freeze()
+    }
+}
+
+/**
+ * Modes for sets.
+ */
+enum class SetMode {
+    UNMANAGED, MANAGED
+}
+
+/**
+ * Supported types by sets. Notice that Mixed sets can in turn support all these types internally
+ * (except Mixed itself).
+ *
+ * Add new types ad-hoc here.
+ */
+enum class SetSupportedType {
+    LONG, INTEGER, SHORT, BYTE, FLOAT, DOUBLE, STRING, BOOLEAN, DATE, DECIMAL128, BINARY, OBJECT_ID,
+    UUID, LINK, MIXED
+}
+
+/**
+ *
+ */
+interface SetTester : GenericTester {
+    override fun setUp(configFactory: TestRealmConfigurationFactory) = Unit     // Not needed here
+    fun isManaged()
```

Comment:
I tried it out that it is supported to mark methods with `@Test` here and trigger the individual test methods when running tests on the implementing class. This means that you can do inheritance by delegation and just do 
```
class ParameterizedSetTests(
        private val tester: SetTester
) : SetTester by tests {
}
``` 
to avoid having to write all the trivial dispatch methods like 
```
   @Test
    fun clear() {
        tester.clear()
    }
```

Reply:
Yeah, I remember you mentioned it in one of the dictionary PRs but I forgot to give it a go. I will try to add this in one of the upcoming PRs.

Reply:
I have created #7362 to address this and the class delegation improvements.